### PR TITLE
Fix lbann half issues

### PR DESCRIPTION
**Pull request by benson31**
This addresses some issues with the HALF library as well as some issues with Copy dispatch. There's also a fix for the crippled timing of the Gemm test.

### CODE REVIEW

File: include/El/blas_like/level1/CopyLocal.hpp [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+}
+
+#ifdef HYDROGEN_HAVE_CUDA
+// If using CUDA, prefer the cudaMemcpy2D implementation. This is
+// ASYNCHRONOUS with respect to the host.
+// (Case 1, GPU)
+//
+// TODO: Profile to verify this is, indeed, faster.
+template <typename T,
+          EnableWhen<IsStorageType<T, Device::CPU>, int> = 0>
```
Comment (timmoon10) [inline include/El/blas_like/level1/CopyLocal.hpp]:
```suggestion
          EnableWhen<IsStorageType<T, Device::GPU>, int> = 0>
```

File: src/blas_like/level1/Copy.cpp [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+void Copy(BaseDistMatrix const& Source, BaseDistMatrix& Target)
+{
+    using FunctorT = details::CopyFunctor;
+    using MatrixTs = ExpandTL<AbstractDistMatrix, MatrixTypes>;
+    using Dispatcher = CopyDispatcher<FunctorT, MatrixTs, MatrixTs>;
+    FunctorT f;
+
+    break_on_me();
+    OutputFromRoot(mpi::COMM_WORLD, "AHHHHHH");
```
Comment (timmoon10) [inline src/blas_like/level1/Copy.cpp]: AHHHHHH

**Review by timmoon10 — APPROVED** [review]
Overall looks good to me, aside from some debugging code left in.

File: src/blas_like/level1/Copy.cpp (hunk 2 above)
Comment (benson31) [reply src/blas_like/level1/Copy.cpp]: I fixed that already. I was using `cout` to help me express my feelings.

Comment (benson31) [reply src/blas_like/level1/Copy.cpp]: And then I changed it back and recommitted it. 🤦‍♂

File: include/El/core/Element/impl.hpp [hunk 3]
```diff
@@ -354,6 +354,19 @@ template<typename Real,
...
+         typename/*=EnableIf<IsStdScalar<F>>*/>
+F Expm1( const F& alpha ) EL_NO_EXCEPT
+{ return std::expm1(alpha); }
+
+template<typename Real,
+         typename/*=EnableIf<IsField<Real>>*/,
+         typename/*=DisableIf<IsStdScalar<Real>>*/,
+         typename/*=void*/>
+Complex<Real> Expm1( const Complex<Real>& alpha ) EL_NO_EXCEPT
+{ return ComplexFromPolar( Expm1(RealPart(alpha)), ImagPart(alpha) ); }
```
Comment (timmoon10) [inline include/El/core/Element/impl.hpp:368]: If we don't properly support a math function for complex types, we should remove it, throw an exception, or return a NaN.

**Review by timmoon10 — APPROVED** [review]

**Review by timmoon10 — APPROVED** [review]

**Pull request merged**